src/ethz_snow/snowing.py

```diff
@@ -0,0 +1,2227 @@
+#  ____     _   _     ___    __        __   _
+# / ___|   | \ | |   / _ \   \ \      / /  (_)   _ __      __ _
+# \___ \   |  \| |  | | | |   \ \ /\ / /   | |  | '_ \    / _` |
+#  ___) |  | |\  |  | |_| |    \ V  V /    | |  | | | |  | (_| |
+# |____/   |_| \_|   \___/      \_/\_/     |_|  |_| |_|   \__, |
+#                                                         |___/
+
+"""Implementation of the SNOWing class.
+
+This module contains the SNOWing class used to run the single vial simulation
+with spatial information.
+"""
+
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import multiprocessing as mp
+import matplotlib.pyplot as plt
+from scipy.integrate import simps
+from typing import Optional
+
+import ethz_snow.utils as Utils
+from ethz_snow.constants import calculateDerived
+from ethz_snow.operatingConditions import OperatingConditions
+
+HEATFLOW_REQUIREDKEYS = ("int", "ext", "s0")
+VIAL_GROUPS = ("corner", "edge", "center", "all")
+
+
+class Snowing:
+    """A class to handle a single SNOWing (Stochastic Nucleation of Water
+     with INternal Gradients) simulation.
+
+    More information regarding the equations and their derivation can be found in
+    XXX, Deck et al. (2023).
+
+    Parameters:
+        k (dict): Heat transfer coefficients.
+        opcond (OperatingConditions): Operating conditions of the run.
+        temperature (str): Model dimensionality.
+        configuration (str): Freezing configuration.
+        plotting (bool): Whether evolutions are plotted or not.
+        Nrep (int): Number of repetitions.
+        configPath (Optional[str]): The path of the (optional) custom config yaml.
+    """
+
+    def __init__(
+        self,
+        k: dict = {"int": 0, "ext": 0, "s0": 50, "s_sigma_rel": 0},
+        opcond: OperatingConditions = OperatingConditions(),
+        temperature: str = "spatial_1D",
+        configuration: str = "shelf",
```

Comment:
I'm not yet convinced it makes sense to be able to provide configPath _and_ `temperature` and `configuration`.

Reply:
The idea of keeping temperature and configuration inputs was that once all the relevant constants are specified in the yaml file, the user could run the model for different configurations and thus with different dimensionalities without having to again modify the yaml file constantly. what do you think?

Reply:
I now removed configuration, dimensionality as additional parameters to snowing. Everything is specified in configPath now. I think you were right, this makes much more sense!